src/modules/ens/sagas.ts

```diff
@@ -207,3 +217,43 @@ function* handleFetchENSListRequest(_action: FetchENSListRequestAction) {
     yield put(fetchENSListFailure(ensError))
   }
 }
+
+function* handleChangeProfile(action: ChangeProfileRequestAction) {
+  try {
+    const { selectedName } = action.payload
+
+    const address = yield select(getAddress)
+
+    const client = new CatalystClient('https://peer.decentraland.org', 'builder')
+    const entities = yield client.fetchEntitiesByPointers(EntityType.PROFILE, [address.toLowerCase()])
+    const profile = entities.length > 0 ? entities[0] : null
+
+    const avatar = profile && profile.metadata && profile.metadata.avatars[0]
+    avatar.name = selectedName
+    avatar.hasClaimedName = true
+
+    // Build entity
+    const content: Map<string, string> = new Map(
+      (profile.content || []).map(({ file, hash }: { file: string; hash: string }) => [file, hash])
+    )
+    const deployPreparationData = yield call(() =>
+      DeploymentBuilder.buildEntityWithoutNewFiles(EntityType.PROFILE, [address], content, profile.metadata)
+    )
+
+    // Request signature
+    const eth = Eth.fromCurrentProvider()
+    if (eth) {
+      const personal = new Personal(eth.provider)
+      const signature = yield personal.sign(deployPreparationData.entityId, address, '')
+
+      // Deploy change
+      const authChain = Authenticator.createSimpleAuthChain(deployPreparationData.entityId, address, signature)
+      yield client.deployEntity({ ...deployPreparationData, authChain })
+
+      yield put(changeProfileSuccess())
```

Comment:
I was going to comment something like this, I would recommend having this saga and action in the `ens` modules, maybe something like `SET_ALIAS_REQUEST/SUCCESS/FAILURE` and within the saga generate a new `Profile` and at the end of the saga dispatch an `UPDATE_PROFILE` action with the new `Profile`, and that action should live in the `profile` module. I'm saying this because the profile module is likely to be moved to `decentraland-dapps` to be used in the marketplace and other dapps, and we'll want the ability to update profiles (thus `UPDATE_PROFILE` should live there) but setting the alias is something related to ENS and problably shouldn't be mixed with the more general profile module.